src/batchpage.js

```diff
@@ -0,0 +1,125 @@
+import Dropdown from './dropdown.js'
+import Navbar from './navbar.js'
+import CollectionsBox from './collections-box.js'
+import Batch from './batch.js'
+import parseQuery from './parseQuery.js'
+
+class BatchPage extends React.Component {
+
+  constructor(props) {
+    super(props)
+    var args = parseQuery(window.location.href)
+    this.state = {
+      terms: args.terms ? ("" + args.terms).replace(/\+/g, ' ') : "",
+      args: args,
+      collections: null,
+      selectedCollections: null,
+      limits: null,
+    }
+  }
+
+  componentDidMount() {
+    this.getCollectionsAndLimits()
+  }
+
+  getCollectionsAndLimits() {
+    $.get('/collections', function (resp) {
+      var collections = resp
+      var selectedCollections = null
+      var args = this.state.args
+      if (args.collections) {
+        var sel = '' + args.collections
+        selectedCollections = sel ? sel.split('+') : []
+      }
+      else {
+        selectedCollections = resp.map((c) => c.name)
+      }
+
+      this.setState({
+        collections,
+        selectedCollections,
+      })
+    }.bind(this))
+
+    $.get('/limits', function(resp) {
+      this.setState({limits: resp})
+    }.bind(this))
+  }
+
+  buildQuery(termsString, selectedCollections, batchSize) {
+    if(! termsString.trim()) return null
+    if(! selectedCollections.length) return null
+
+    return {
+      terms: termsString.trim().split('\r\n'),
+      collections: selectedCollections,
+      batchSize: batchSize,
+    }
+  }
+
+  render() {
+    if(! (this.state.collections && this.state.limits)) {
+      return <p>loading ...</p>
+    }
+
+    let refreshForm = () => {
+      if(this.refs.terms.value)
+        this.refs.form.submit()
+    }
+
+    let onChangeCollections = (selected) => {
+      this.setState({selectedCollections: selected})
+      setTimeout(refreshForm, 0)
+    }
+
+    let collectionsValue = this.state.selectedCollections.join(' ')
+    let {terms, collections, selectedCollections, limits} = this.state
+    let batchSize = limits.batch
+    let query = this.buildQuery(terms, selectedCollections, batchSize)
+
+    return (
+      <form id="batch-form" ref="form">
+        <input type="hidden" name="collections" value={collectionsValue} />
+        <div className="row">
+          <div className="col-sm-2">
+            <h1>Hoover</h1>
+            <CollectionsBox
+              collections={collections}
+              selected={selectedCollections}
+              onChange={onChangeCollections} />
+          </div>
+          <div className="col-sm-9">
+            <div id="batch-input-box" className="form-group">
+              <textarea
+                id="batch-input-terms"
+                ref="terms"
+                name="terms"
+                className="form-control"
+                rows="8"
+                placeholder="search terms, one per line"
+                defaultValue={terms}
+                ></textarea>
+            </div>
```

Comment:
There should be some counter here for how many query lines the user has inputed in the textarea (so they can figure out why they're being limited and how many lines they have to remove)